### PR TITLE
Add ability to override images in docker registry

### PR DESCRIPTION
**Pull request by johnlhamilton**
Fixes CC-1997

### CODE REVIEW

File: cli/cmd/docker.go [hunk 1]
```diff
@@ -44,6 +47,11 @@ func (c *ServicedCli) initDocker() {
 					cli.StringFlag{"registry", "", "host:port where the registry is running"},
 					cli.BoolFlag{"override, f", "overrides all existing image records"},
 				},
+			}, {
+				Name:        "override",
+				Usage:       "Replace an image in the registry with a new image",
+				Description: "serviced docker override NEWIMAGE OLDIMAGE",
```
Comment (kwalkerz) [inline cli/cmd/docker.go]: To match typical Unix command style, should we switch this to "OLDIMAGE NEWIMAGE". cp, mv, ln, even diff kind of, all use this pattern.

Comment (johnlhamilton) [inline cli/cmd/docker.go]: Yeah, that would be a little more intuitive.  I'll make that change

File: dfs/override.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	newImage, err := dfs.docker.FindImage(newimg)
+	if err != nil {
+		glog.Errorf("Could not find replacement image %s: %s", newimg, err)
+		return err
+	}
+
+	// push the image into the registry
+	hash, err := dfs.docker.GetImageHash(newImage.ID)
+	if err != nil {
+		glog.Errorf("Could not get has for image %s: %s", newimg, err)
```
Comment (kwalkerz) [inline dfs/override.go]: Missing h in "hash"

File: dfs/override.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		return err
+	}
+
+	if err := dfs.index.PushImage(oldImage.String(), newImage.ID, hash); err != nil {
+		glog.Errorf("Could not push image %s (%s): %s", oldImage, newImage.ID, err)
```
Comment (kwalkerz) [inline dfs/override.go]: Can we make this error message a little clearer? More along the lines "Could not replace image A with B"...

File: facade/dfs.go [hunk 4]
```diff
@@ -637,3 +637,9 @@ func (info *registryVersionInfo) start(isvcsRoot string, hostPort string) (*dock
 
 	return container, nil
 }
+
+// DockerOverride will replace a docker image in the registry with a new image for all services that use
+//  it under the given tenant
```
Comment (kwalkerz) [inline facade/dfs.go]: Is this comment accurate? There's nothing in the API that limits the image to a specific tenant...

Comment (johnlhamilton) [inline facade/dfs.go]: Nope, leftover from an earlier approach.  I'll remove the part about a given tenant.

**Pull request merged**